src/tribler/core/components/libtorrent/download_manager/download_manager.py

```diff
@@ -363,7 +363,8 @@ def set_max_connections(self, conns, hops=None):
     def set_upload_rate_limit(self, rate, hops=None):
         # Rate conversion due to the fact that we had a different system with Swift
         # and the old python BitTorrent core: unlimited == 0, stop == -1, else rate in kbytes
-        libtorrent_rate = int(-1 if rate == 0 else (1 if rate == -1 else rate * 1024))
+        torrent_rate = {0: -1, -1: 1}
```

Comment:
I do plan to add a test before marking it as ready. Like I said here https://github.com/Tribler/tribler/pull/7466#issuecomment-1614588631, I converted it from "draft" to "ready" by accident 😅.

I changed it to use a dictionary first to make it clearer, and second, to avoid branching. A series of `if` statements like you wrote is clearer than what I wrote, but in return it branches. I don't mind rewriting it like you suggested. Which one do you prefer?

Reply:
> I changed it to use a dictionary first to make it clearer, and second, to avoid branching. A series of if statements like you wrote is clearer than what I wrote, but in return it branches. I don't mind rewriting it like you suggested. Which one do you prefer?

I lean towards the version that incorporates branches. As far as I can see, there don't appear to be any disadvantages to using them.

> The first converts 0 => -1, -1 => 1. The second does so in reverse, meaning -1 => 0, 1 => -1. 
... 
Which option should I go with?

I believe option (1) seems to be the most straightforward and uncomplicated, so I would favor it over the other options.

---

Concerning the preference of using a dictionary over branching — we could consider seeking a second opinion on this matter.

@kozlovsky what version would you prefer?

1:
```python
    @staticmethod
    def convert_rate(rate):
        # Rate conversion due to the fact that we had a different system with Swift
        # and the old python BitTorrent core: unlimited == 0, stop == -1, else rate in kbytes
        if rate == -1:
            return 1
        if rate == 0:
            return -1
        return rate * 1024
```


2:
```python
    @staticmethod
    def convert_rate(rate):
        # Rate conversion due to the fact that we had a different system with Swift
        # and the old python BitTorrent core: unlimited == 0, stop == -1, else rate in kbytes
        torrent_rate = {0: -1, -1: 1}
        return torrent_rate.get(rate, rate * 1024)
```

Reply:
Usually, I prefer the shorter option for simple expressions, but in this case, the first option with ifs looks clearer to me; it is easier to understand. So, I vote for the first option :)